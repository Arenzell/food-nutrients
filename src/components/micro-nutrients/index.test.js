import React from 'react'
import MicroNutrients from './index'
import renderer from 'react-test-renderer'

it('renders correctly with micro nutrients data', () => {
  const nutrients = [
    {
      name: 'Vitamin A',
      rda: 900,
      wiki: 'https://en.wikipedia.org/wiki/Vitamin_A',
      required: true,
      type: 'Vitamins',
      tui: 3000,
    },
    {
      name: 'Chloride',
      rda: 2300000,
      wiki: 'https://en.wikipedia.org/wiki/Chloride',
      required: true,
      type: 'Minerals',
      tui: 15000000,
    },
  ]
  const microNutrients = {
    'Vitamin A': {
      nutrient: {
        name: 'Vitamin A',
        rda: 900,
        wiki: 'https://en.wikipedia.org/wiki/Vitamin_A',
        required: true,
        type: 'Vitamins',
        tui: 3000,
      },
      raw: 50,
      amount: 5,
      percentage: 30,
      unit: 'mg',
    },
    Chloride: {
      nutrient: {
        name: 'Chloride',
        rda: 2300000,
        wiki: 'https://en.wikipedia.org/wiki/Chloride',
        required: true,
        type: 'Minerals',
        tui: 15000000,
      },
      raw: 100,
      amount: 10,
      percentage: 40,
      unit: 'mg',
    },
  }
  const tree = renderer
    .create(<MicroNutrients definitions={nutrients} microNutrients={microNutrients} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly with null nutrients data', () => {
  const nutrients = [
    {
      name: 'Vitamin A',
      rda: 900,
      wiki: 'https://en.wikipedia.org/wiki/Vitamin_A',
      required: true,
      type: 'Vitamins',
      tui: 3000,
    },
    {
      name: 'Chloride',
      rda: 2300000,
      wiki: 'https://en.wikipedia.org/wiki/Chloride',
      required: true,
      type: 'Minerals',
      tui: 15000000,
    },
  ]
  const microNutrients = {}
  const tree = renderer
    .create(<MicroNutrients definitions={nutrients} microNutrients={microNutrients} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
