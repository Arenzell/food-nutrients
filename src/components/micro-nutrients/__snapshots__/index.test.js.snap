// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with micro nutrients data 1`] = `
<div
  className="microNutrients"
>
  <div
    className="microNutrientType"
  >
    <span
      className="nutrientType"
    >
      Vitamins
    </span>
    <div
      className="microNutrient"
    >
      <div
        className="microNutrientName"
      >
        <a
          href="https://en.wikipedia.org/wiki/Vitamin_A"
          rel="noopener noreferrer"
          target="_blank"
        >
          Vitamin A
        </a>
      </div>
      <div
        className="microNutrientProgressBar"
      >
        <div
          className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-small"
        >
          <div>
            <div
              className="ant-progress-outer"
            >
              <div
                className="ant-progress-inner"
              >
                <div
                  className="ant-progress-bg"
                  style={
                    Object {
                      "background": undefined,
                      "borderRadius": "100px",
                      "height": 6,
                      "width": "30%",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <span>
          30
          % 
        </span>
      </div>
      <div
        className="microNutrientAmount"
      >
        5
         
        mg
      </div>
    </div>
  </div>
  <div
    className="microNutrientType"
  >
    <span
      className="nutrientType"
    >
      Minerals
    </span>
    <div
      className="microNutrient"
    >
      <div
        className="microNutrientName"
      >
        <a
          href="https://en.wikipedia.org/wiki/Chloride"
          rel="noopener noreferrer"
          target="_blank"
        >
          Chloride
        </a>
      </div>
      <div
        className="microNutrientProgressBar"
      >
        <div
          className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-small"
        >
          <div>
            <div
              className="ant-progress-outer"
            >
              <div
                className="ant-progress-inner"
              >
                <div
                  className="ant-progress-bg"
                  style={
                    Object {
                      "background": undefined,
                      "borderRadius": "100px",
                      "height": 6,
                      "width": "40%",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <span>
          40
          % 
        </span>
      </div>
      <div
        className="microNutrientAmount"
      >
        10
         
        mg
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly with null nutrients data 1`] = `
<div
  className="microNutrients"
>
  <div
    className="microNutrientType"
  >
    <span
      className="nutrientType"
    >
      Vitamins
    </span>
    <div
      className="microNutrient"
    >
      <div
        className="microNutrientName"
      >
        <a
          href="https://en.wikipedia.org/wiki/Vitamin_A"
          rel="noopener noreferrer"
          target="_blank"
        >
          Vitamin A
        </a>
      </div>
      <div
        className="microNutrientProgressBar"
      >
        <div
          className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-small"
        >
          <div>
            <div
              className="ant-progress-outer"
            >
              <div
                className="ant-progress-inner"
              >
                <div
                  className="ant-progress-bg"
                  style={
                    Object {
                      "background": undefined,
                      "borderRadius": "100px",
                      "height": 6,
                      "width": "0%",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <span>
          0
          % 
        </span>
      </div>
      <div
        className="microNutrientAmount"
      >
        0
         
        μg
      </div>
    </div>
  </div>
  <div
    className="microNutrientType"
  >
    <span
      className="nutrientType"
    >
      Minerals
    </span>
    <div
      className="microNutrient"
    >
      <div
        className="microNutrientName"
      >
        <a
          href="https://en.wikipedia.org/wiki/Chloride"
          rel="noopener noreferrer"
          target="_blank"
        >
          Chloride
        </a>
      </div>
      <div
        className="microNutrientProgressBar"
      >
        <div
          className="ant-progress ant-progress-line ant-progress-status-normal ant-progress-small"
        >
          <div>
            <div
              className="ant-progress-outer"
            >
              <div
                className="ant-progress-inner"
              >
                <div
                  className="ant-progress-bg"
                  style={
                    Object {
                      "background": undefined,
                      "borderRadius": "100px",
                      "height": 6,
                      "width": "0%",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
        <span>
          0
          % 
        </span>
      </div>
      <div
        className="microNutrientAmount"
      >
        0
         
        μg
      </div>
    </div>
  </div>
</div>
`;
